corehq/apps/repeaters/models.py

```diff
@@ -593,6 +593,10 @@ def _fail(self, reason, response):
             'target_url': self.url,
         })
 
+    def add_failure_reason(self, failure_reason):
```

Comment:
As far my memory goes, i wanted to save error message on repeat record instead of case as done [currently](https://github.com/dimagi/commcare-hq/blob/master/custom/enikshay/integrations/ninetyninedots/repeater_generators.py#L63), so made this. And I don't think we need to retry in this case. This should be serving request errors which would recur on re-request.